src/main/java/com/example/bookstoreapp/dto/user/UserRegistrationRequestDto.java

```diff
@@ -0,0 +1,30 @@
+package com.example.bookstoreapp.dto.user;
+
+import com.example.bookstoreapp.validation.FieldMatch;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+
+@FieldMatch(message = "Password don`t match",
+        field = "password",
```

Comment:
you can simplify by @FieldMatch(fields = {"password", "repeatPassword"}, message.... )